lib/bump/dependency_file_updaters/ruby.rb

```diff
@@ -94,9 +95,18 @@ def build_updated_gemfile_lock
           end
         post_process_lockfile(lockfile_body)
       rescue SharedHelpers::ChildProcessFailed => error
-        raise unless error.error_class == "Bundler::VersionConflict"
+        handle_bundler_errors(error)
+      end
 
-        raise DependencyFileUpdaters::VersionConflict
+      def handle_bundler_errors(error)
+        case error.error_class
+        when "Bundler::VersionConflict"
+          raise DependencyFileUpdaters::VersionConflict
+        when "Bundler::Source::Git::GitCommandError"
+          raise Bump::GitDependencyNotFound
```

Comment:
This could get confusing if the git command fails for other reasons. Is there anything the exception gives us that indicates it's a not found error vs something else?